### DISCUSSION TITLE
StrongSwan and Cisco Unity router

### DISCUSSION THREAD

**Discussion by tjmcclure0501** [strongswan/strongswan#280] [Q&A]
I am struggling trying to get Strong Swan to connect a customer environment that is using a Cisco Unity router.  I have everything configured correctly but I keep getting a hash mismatch error.  I think the issue is I am trying to integrate with old equipment.  I noticed this web page:
https://wiki.strongswan.org/projects/strongswan/wiki/UnityPlugin
How do I enable this plugin?  Do I have to recompile Strong Swan?  Does it make sense this could be the reason I am struggling?  I will work with the customer to upgrade but first I need to get integrated with them.  
I installed these plugins: libstrongswan-standard-plugins libstrongswan-extra-plugins

I  include them in strongswan.conf as follows:
        plugins {
                include strongswan.d/charon/*.conf
        }
        include strongswan.d/*.conf

Any help will be greatly appreciated.

added this to strongswan.conf:  cisco_unity = yes
did not help
Thanks,
Tim

**Comment by strongX509** [comment]
Tim,
without a strongSwan logfile it is very difficult to analyze your problem. If the Cisco router suddenly discontinues the negotiation then
setting
```
charon {
  cisco_unity = yes
}
```
in `/etc/strongswan.conf` might help.

Best regards
Andreas
